plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -783,8 +813,118 @@ public function create( &$order ) {
 		throw new \Exception( 'Unimplemented' );
 	}
 
+	/**
+	 * Method to update an order in the database.
+	 *
+	 * @param \WC_Order $order
+	 */
 	public function update( &$order ) {
-		throw new \Exception( 'Unimplemented' );
+		global $wpdb;
+
+		$order->set_version( Constants::get_constant( 'WC_VERSION' ) );
+
+		// Before updating, ensure date paid is set if missing.
+		if ( ! $order->get_date_paid( 'edit' ) && version_compare( $order->get_version( 'edit' ), '3.0', '<' ) && $order->has_status( apply_filters( 'woocommerce_payment_complete_order_status', $order->needs_processing() ? 'processing' : 'completed', $order->get_id(), $order ) ) ) { // phpcs:ignore WooCommerce.Commenting.CommentHooks.HookCommentWrongStyle
```

Comment:
let's split them to different lines (or a separate function) for better readability.

Reply:
I think this was missed